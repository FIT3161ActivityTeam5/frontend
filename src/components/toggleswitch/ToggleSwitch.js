import React, { Component } from "react";
import './ToggleSwitch.scss';

class ToggleSwitch extends Component {
    render() {
        return (
            <div className="toggle-switch">
                <input
                    type="checkbox"
                    className="toggle-switch-checkbox"
                    name="toggleSwitch"
                    id="toggleSwitch"
                />
                <label className="toggle-switch-label" htmlFor={this.props.Name}>
                    <span className="toggle-switch-inner" data-yes="Ja" data-no="Nein" />
                    <span className="toggle-switch-switch" />
                </label>
            </div>
        );
    }
}

export default ToggleSwitch;
