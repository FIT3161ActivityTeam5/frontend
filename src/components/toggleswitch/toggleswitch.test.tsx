import React from 'react';
import renderer from 'react-test-renderer';
import ToggleSwitch from './toggleswitch';

describe('<Button />', () => {
    // Snapshot test.
    it('renders correctly', () => {
        const tree = renderer.create(<ToggleSwitch />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom styles correctly', () => {
        const tree = renderer.create(
            <ToggleSwitch/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom text styles correctly', () => {
        const tree = renderer.create(
            <ToggleSwitch />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});