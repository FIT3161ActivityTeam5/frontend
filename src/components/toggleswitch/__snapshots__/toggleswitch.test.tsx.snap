// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> applies custom styles correctly 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="rgba(124, 58, 237, 1)"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Object {
          "justifyContent": "flex-end",
          "textAlign": "justify",
        },
        Object {
          "backgroundColor": "#3e3e3e",
          "borderRadius": 16,
        },
      ],
    ]
  }
  thumbTintColor="rgba(245, 243, 255, 1)"
  tintColor="rgba(124, 58, 237, 1)"
  value={false}
/>
`;

exports[`<Button /> applies custom text styles correctly 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="rgba(124, 58, 237, 1)"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Object {
          "justifyContent": "flex-end",
          "textAlign": "justify",
        },
        Object {
          "backgroundColor": "#3e3e3e",
          "borderRadius": 16,
        },
      ],
    ]
  }
  thumbTintColor="rgba(245, 243, 255, 1)"
  tintColor="rgba(124, 58, 237, 1)"
  value={false}
/>
`;

exports[`<Button /> renders correctly 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="rgba(124, 58, 237, 1)"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Object {
          "justifyContent": "flex-end",
          "textAlign": "justify",
        },
        Object {
          "backgroundColor": "#3e3e3e",
          "borderRadius": 16,
        },
      ],
    ]
  }
  thumbTintColor="rgba(245, 243, 255, 1)"
  tintColor="rgba(124, 58, 237, 1)"
  value={false}
/>
`;
